specification/communication/data-plane/CallingServer/preview/2021-06-15-preview/communicationservicescallingserver.json

```diff
@@ -0,0 +1,1467 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2021-06-15-preview",
+    "title": "Azure Communication CallingServer Service",
+    "description": "Azure Communication CallingServer Service"
+  },
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/calling/callConnections": {
+      "post": {
+        "tags": [
+          "CallConnection"
+        ],
+        "summary": "Create a new call.",
+        "description": "Create a new call.",
+        "operationId": "CallConnection_CreateCall",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "callRequest",
+            "in": "body",
+            "description": "Create call request.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CreateCallRequest"
+            }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Returns the create call response.",
+            "schema": {
+              "$ref": "#/definitions/CreateCallResult"
+            }
+          },
+          "401": {
+            "description": "Unauthorized",
+            "schema": {
+              "$ref": "#/definitions/CommunicationError"
+            },
+            "x-ms-error-response": true
+          },
+          "400": {
+            "description": "Bad request",
+            "schema": {
+              "$ref": "#/definitions/CommunicationError"
+            },
+            "x-ms-error-response": true
+          },
+          "500": {
+            "description": "Internal server error.",
+            "schema": {
+              "$ref": "#/definitions/CommunicationError"
+            },
+            "x-ms-error-response": true
+          }
+        }
+      }
+    },
+    "/calling/callConnections/{callConnectionId}/:hangup": {
```

Comment:
I find it very likely that we are going to want to be able to handle the race condition of hanging up/at the same time as the call was being answered. In other words, differentiating between giving up due to no answer and hanging up an active connection. We should make room for this in the API.

Reply:
If the callee doesn't answer the call, after a timeout, the call will go to Disconnected state. If the app wants to hangup while the call is still ringing, app can call hangup() and we will handle that internally. Do you see a benefit for the developer in differentiating between these two cases? Could you suggest what they would differently.

Reply:
You allow application developers to avoid exposing called parties the annoyance of answering and hearing the other party hang up. It is my recollection (which is about 2 decades old by this time) that all libraries we used distinguished between the two. And as I look at the code `CreateCallRequest`, I am also missing a timeout parameter that allows the caller to determine for how long they want to wait for the destination to "pick up the phone".